charts/redpanda/templates/service.internal.yaml

```
@@ -16,14 +16,15 @@ limitations under the License.
 */}}
 ---
 # This service is only used to create the DNS enteries for each pod in
-# the stateful set. This service should not be used by any client
-# application
+# the stateful set and allow the serviceMonitor to target the pods.
+# This service should not be used by any client application
 apiVersion: v1
 kind: Service
 metadata:
   name: {{ include "redpanda.fullname" . }}
   namespace: {{ .Release.Namespace | quote }}
   labels:
+    monitoring/enabled: "{{ .Values.monitoring.enabled }}"
```

To distinguish labels that are specific to our use case, it's good practice to add a domain.

```suggestion
    monitoring.redpanda.com/enabled: "{{ .Values.monitoring.enabled }}"
```